lix/source-code/client/src/openRepository.ts

```diff
@@ -92,17 +107,49 @@ export async function openRepository(
 
 	await pending
 
-	// delay all fs and repo operations until the repo clone and checkout have finished, this is preparation for the lazy feature
-	function delayedAction({ execute }: { execute: () => any }) {
-		if (pending) {
-			return pending.then(execute)
-		}
+	const oidToFilePaths = {} as { [oid: string]: string[] }
+	const filePathToOid = {} as { [filePath: string]: string }
```

Comment:
replace cached mapping and use either walk ad hoc, or something like raw.listFiles or similar